xbmc/cores/dvdplayer/DVDCodecs/Video/DXVA.cpp

```diff
@@ -238,9 +238,10 @@ STDMETHODIMP_(ULONG) CDXVADecoderWrapper::AddRef(void)
 STDMETHODIMP_(ULONG) CDXVADecoderWrapper::Release(void)
 {
   AtomicDecrement(&m_refs);
-  assert(m_refs >= 0);
-  if (m_refs == 0) delete this;
-  return m_refs;
+  volatile long refs = m_refs;
```

Comment:
Original isn't thread safe either but you can stay with original in this pr, it's a bit tricky to get right and I need to test it out